components/lib/crash/src/test/java/mozilla/components/lib/crash/handler/CrashHandlerServiceTest.kt

```diff
@@ -48,52 +54,54 @@ class CrashHandlerServiceTest {
         )).install(testContext)
 
         intent = Intent("org.mozilla.gecko.ACTION_CRASHED")
-        intent?.component = ComponentName(
+        intent!!.component = ComponentName(
             "org.mozilla.samples.browser",
             "mozilla.components.lib.crash.handler.CrashHandlerService"
         )
-        intent?.putExtra(
+        intent!!.putExtra(
             "uuid",
             "94f66ed7-50c7-41d1-96a7-299139a8c2af"
         )
-        intent?.putExtra(
+        intent!!.putExtra(
             "minidumpPath",
             "/data/data/org.mozilla.samples.browser/files/mozilla/Crash Reports/pending/3ba5f665-8422-dc8e-a88e-fc65c081d304.dmp"
         )
-        intent?.putExtra(
+        intent!!.putExtra(
             "extrasPath",
             "/data/data/org.mozilla.samples.browser/files/mozilla/Crash Reports/pending/3ba5f665-8422-dc8e-a88e-fc65c081d304.extra"
         )
-        intent?.putExtra("minidumpSuccess", true)
-        service = spy(CrashHandlerService())
+        intent!!.putExtra("minidumpSuccess", true)
     }
 
     @After
     fun tearDown() {
+        service!!.stopService(Intent())
         CrashReporter.reset()
     }
 
     @Test
     fun `CrashHandlerService forwards fatal native code crash to crash reporter`() {
-        doNothing().`when`(reporter)?.sendCrashReport(any(), any())
-        doNothing().`when`(service)?.kill()
+        doNothing().`when`(reporter)!!.sendCrashReport(any(), any())
 
-        intent?.putExtra("fatal", true)
-        service?.onHandleIntent(intent)
-        verify(reporter)?.onCrash(any(), any())
-        verify(reporter)?.sendCrashReport(any(), any())
-        verify(reporter, never())?.sendNonFatalCrashIntent(any(), any())
+        CoroutineScope(IO).launch {
+            intent!!.putExtra("fatal", true)
+            service!!.onStartCommand(intent!!, 0, 0)
+            verify(reporter)!!.onCrash(any(), any())
+            verify(reporter)!!.sendCrashReport(any(), any())
+            verify(reporter, never())!!.sendNonFatalCrashIntent(any(), any())
+        }
```

Comment:
Should this be `runBlocking` in the test? Otherwise this will likely run after the test method has already completed and never actually get verified. And even if it fails then it would crash in the thread of the coroutine and not here.

Reply:
My mistake.  I'll update so tests can inject their own scope.